src/libraries/System.Linq/src/System/Linq/SequenceEqual.cs

```diff
@@ -29,42 +30,57 @@ public static bool SequenceEqual<TSource>(this IEnumerable<TSource> first, IEnum
                     return ((ReadOnlySpan<TSource>)firstArray).SequenceEqual(secondArray, comparer);
                 }
 
+                if (first is List<TSource> firstList && second is List<TSource> secondList)
+                {
+                    return CollectionsMarshal.AsSpan(firstList).SequenceEqual(CollectionsMarshal.AsSpan(secondList), comparer);
```

Comment:
This should be using TryGetSpan instead of the separate array and list checks.

Reply:
Thanks, I missed that method and it makes it even better since `T[] && List<T>` succeed such check.